### PR TITLE
chore: use new forge feature for filtering coverage

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Remove lcov toolchain from coverage script in favor of new native forge features
- Require strictly increasing project coverage

### Drive-by changes

- Add coverage gutters to vscode recommendations for inline view

### Backward compatibility

Yes

### Testing

See codecov report comment

### CODE REVIEW

File: solidity/coverage.sh [hunk 1]
```diff
@@ -3,29 +3,10 @@
 # exit on error
 set -e
 
-# generates lcov.info
+# exclude FastTokenRouter until https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2806 is resolved
```
Comment (aroralanuk) [inline solidity/coverage.sh:6]: neat

File: codecov.yml [hunk 2]
```diff
@@ -1,3 +1,10 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto  # strictly increasing coverage
+        threshold: 5% # buffer for coverage drop
```
Comment (aroralanuk) [inline codecov.yml:6]: Isn't 5% a bit high? 5% * 1890 = 94.5
3% is 56.7 which seems more apt?

(assuming consistency between runs)

**Review by aroralanuk — APPROVED** [review]
LGTM

File: codecov.yml (hunk 2 above)
Comment (yorhodes) [reply codecov.yml:6]: it is pretty high but left it large for flakiness

Comment (yorhodes) [inline codecov.yml:6]:
```suggestion
        threshold: 3% # buffer for coverage drop
```

**Pull request merged**